src/pharmpy/model/external/nonmem/update.py

```diff
@@ -1587,7 +1587,13 @@ def update_estimation(control_stream, model):
     old = model.internals.old_estimation_steps
     new = model.estimation_steps
     if old == new:
-        return control_stream
+        if (
+            len(old) > 0
+            and len(new) > 0
+            and old[-1].predictions == new[-1].predictions
+            and old[-1].residuals == new[-1].residuals
+        ):
```

Comment:
Why is this needed?  If  `old == new` then all of these must also be true. Or is this check missing in `__eq__` for `EstimationStep` ?

Reply:
these are missing in `__eq__` for `EstimationStep`. Shall I add it there instead?